azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -275,62 +290,95 @@ private FlowRunner createFlowRunner(final int execId) throws ExecutorManagerExce
+  /**
+   * Submits the flow to executorService for execution.
+   * @param flowRunner The FlowRunner object.
+   */
+  private void submitFlowRunner(final FlowRunner flowRunner) throws ExecutorManagerException {
     // set running flow, put it in DB
-    this.flowFuture = this.executorService.submit(flowRunner);
+    logger.info("Submitting flow with execution Id " + flowRunner.getExecutionId());
+    final Future<?> flowFuture = this.executorService.submit(flowRunner);
     try {
-      this.flowFuture.get();
-    } catch (final InterruptedException ie) {
-      ie.printStackTrace();
-    } catch (final ExecutionException ee) {
-      ee.printStackTrace();
+      flowFuture.get();
+    } catch (final InterruptedException | ExecutionException e) {
+      logger.error(ExceptionUtils.getStackTrace(e));
+      throw new ExecutorManagerException(e);
     }
   }
 
+  /**
+   * Unzip a file.
+   * @param zipPath The source zip file.
+   * @param dirPath The destination of the zip file content.
+   * @throws IOException
+   */
   private static void unzipFile(final String zipPath,
       final Path dirPath) throws IOException {
     final ZipFile zipFile = new ZipFile(zipPath);
-    System.out.println("source path = " + zipPath);
+    logger.info("source path = " + zipPath);
     final File unzipped = new File(dirPath.toString());
-    System.out.println("unzipped file dir = " + unzipped.toString());
+    logger.info("unzipped file dir = " + unzipped.toString());
     Utils.unzip(zipFile, unzipped);
     zipFile.close();
   }
 
-  private ExecutorLoader setupDao(final DatabaseOperator dbOperator) {
-    final ExecutionFlowDao executionFlowDao = new ExecutionFlowDao(dbOperator, null);
-    final ExecutorDao executorDao = new ExecutorDao(dbOperator);
-    final ExecutionJobDao executionJobDao = new ExecutionJobDao(dbOperator);
-    final ExecutionLogsDao executionLogsDao = new ExecutionLogsDao(dbOperator);
-    final ExecutorEventsDao executorEventsDao = new ExecutorEventsDao(dbOperator);
-    final ActiveExecutingFlowsDao activeExecutingFlowsDao =
-        new ActiveExecutingFlowsDao(dbOperator);
-    final FetchActiveFlowDao fetchActiveFlowDao =
-        new FetchActiveFlowDao(dbOperator);
-    final AssignExecutorDao assignExecutorDao =
-        new AssignExecutorDao(dbOperator, executorDao);
-    final NumExecutionsDao numExecutionsDao = new NumExecutionsDao(dbOperator);
-    final ExecutionRampDao executionRampDao = new ExecutionRampDao(dbOperator);
-
-    // Use above objects to create Executor Loader
-    return new JdbcExecutorLoader(executionFlowDao, executorDao,
-        executionJobDao, executionLogsDao, executorEventsDao,
-        activeExecutingFlowsDao, fetchActiveFlowDao, assignExecutorDao,
-        numExecutionsDao, executionRampDao);
+  /**
+   * Setup in-memory keystore to be reused for all the job executions in the flow.
+   * @throws IOException
+   */
+  private void setupKeyStore() throws IOException {
```

Comment:
What if we don't want to use in memory keystore? Can you please keep it configurable?